edx_proctoring/views.py

```diff
@@ -1104,3 +1107,40 @@ def post(self, request, user_id):  # pylint: disable=unused-argument
                         code = 500
                 seen.add(backend_name)
         return Response(data=results, status=code)
+
+
+class UserRetirement(AuthenticatedAPIView):
+    """
+    Retire user personally-identifiable information (PII) for a user
+    """
+    def _retire_exam_attempts_user_info(self, user_id):
+        """ Obfuscate PII for exam attempts and exam history """
+        attempts = ProctoredExamStudentAttempt.objects.filter(user_id=user_id)
+        if attempts:
+            for attempt in attempts:
+                attempt.student_name = obscured_user_id(attempt.student_name)
+                attempt.last_poll_ipaddr = obscured_user_id(attempt.last_poll_ipaddr)
+                attempt.save()
+
+        attempts_history = ProctoredExamStudentAttemptHistory.objects.filter(user_id=user_id)
+        if attempts_history:
+            for attempt_history in attempts_history:
+                attempt_history.student_name = obscured_user_id(attempt_history.student_name)
+                attempt_history.last_poll_ipaddr = obscured_user_id(attempt_history.last_poll_ipaddr)
+                attempt_history.save()
+
+    def _retire_user_allowances(self, user_id):
+        """ Delete user allowances """
+        ProctoredExamStudentAllowance.objects.filter(user=user_id).delete()
+        ProctoredExamStudentAllowanceHistory.objects.filter(user=user_id).delete()
+
+    def post(self, request, user_id):  # pylint: disable=unused-argument
+        """ Obfuscates all PII for a given user_id """
+        if not request.user.has_perm('accounts.can_retire_user'):
```

Comment:
In most, if not all, other places we allow admin users to make the retirement calls as well. Might be nice to add just for consistency.

Reply:
@bmedx, do you have the permission string handy? I'm happy to add it.

Reply:
This actually looks like it's been cleaned up to happen automatically, so you should be set! https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/user_api/rules.py